rest/config.go

```diff
@@ -1454,6 +1458,10 @@ func (sc *ServerContext) FetchConfigs(ctx context.Context, isInitialStartup bool
 		//		i++
 		//	}
 		// }
+		err := sc.BootstrapContext.Connection.SetConnectionStringServerless()
+		if err != nil {
+			return nil, err
+		}
```

Comment:
When running through on the dubugger there was times when we hit the function below `FetchDatabases` and the connection string was simple just couchbase://localhost with no serverless connection options so that initial cluster connection was using the normal buffer. This was my way around that. I saw your comment about adding this in `NewClusterConnection ` but I wasn't sure where I would get a serverless flag from there to ensure we only set it when runnning in serverless.